lib/extractor/layer.ts

```diff
@@ -53,6 +56,73 @@ export async function extractDockerArchive(
   });
 }
 
+export async function extractOciArchive(
```

Comment:
feels to me like this should either be some how merged with ```extractDockerArchive```
or we want this module/library refactored so there's different types of extractors, one for Docker and one for OCI images.
either way, let's try to reduce the cost of change here, taking into that the layer scan and extraction process may change in the future to support layer scanning & caching.
wdyt?

Reply:
That's actually a good idea, to have separate extractors. I'm following up this PR with another task, which is doing a fair bit of refactoring (because it will be using this code) so it makes sense to make the changes there. I will definitely address that.